In [4]:
#importing necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [5]:
#importing the dataset
churn_data = pd.read_csv('/Users/jeevithaselvaraj/Downloads/Bank_churn_modelling.csv')

In [6]:
# To understand the data
churn_data

CustomerId    Surname  CreditScore Geography  Gender  Age  Tenure  \
0       15634602   Hargrave          619    France  Female   42       2   
1       15647311       Hill          608     Spain  Female   41       1   
2       15619304       Onio          502    France  Female   42       8   
3       15701354       Boni          699    France  Female   39       1   
4       15737888   Mitchell          850     Spain  Female   43       2   
...          ...        ...          ...       ...     ...  ...     ...   
9995    15606229   Obijiaku          771    France    Male   39       5   
9996    15569892  Johnstone          516    France    Male   35      10   
9997    15584532        Liu          709    France  Female   36       7   
9998    15682355  Sabbatini          772   Germany    Male   42       3   
9999    15628319     Walker          792    France  Female   28       4   

        Balance  Num Of Products  Has Credit Card  Is Active Member  \
0          0.00                1                1                 1   
1      83807.86                1                0                 1   
2     159660.80                3                1                 0   
3          0.00                2                0                 0   
4     125510.82                1                1                 1   
...         ...              ...              ...               ...   
9995       0.00                2                1                 0   
9996   57369.61                1                1                 1   
9997       0.00                1                0                 1   
9998   75075.31                2                1                 0   
9999  130142.79                1                1                 0   

      Estimated Salary  Churn  
0            101348.88      1  
1            112542.58      0  
2            113931.57      1  
3             93826.63      0  
4             79084.10      0  
...                ...    ...  
9995          96270.64      0  
9996         101699.77      0  
9997          42085.58      1  
9998          92888.52      1  
9999          38190.78      0  

[10000 rows x 13 columns]

In [13]:
# to understand the data checking the first 5 rows of the dataset
churn_data.head()

CustomerId   Surname  CreditScore Geography  Gender  Age  Tenure  \
0    15634602  Hargrave          619    France  Female   42       2   
1    15647311      Hill          608     Spain  Female   41       1   
2    15619304      Onio          502    France  Female   42       8   
3    15701354      Boni          699    France  Female   39       1   
4    15737888  Mitchell          850     Spain  Female   43       2   

     Balance  Num Of Products  Has Credit Card  Is Active Member  \
0       0.00                1                1                 1   
1   83807.86                1                0                 1   
2  159660.80                3                1                 0   
3       0.00                2                0                 0   
4  125510.82                1                1                 1   

   Estimated Salary  Churn  
0         101348.88      1  
1         112542.58      0  
2         113931.57      1  
3          93826.63      0  
4          79084.10      0

In [14]:
# to check how many rows in the datset 
churn_data.tail()

CustomerId    Surname  CreditScore Geography  Gender  Age  Tenure  \
9995    15606229   Obijiaku          771    France    Male   39       5   
9996    15569892  Johnstone          516    France    Male   35      10   
9997    15584532        Liu          709    France  Female   36       7   
9998    15682355  Sabbatini          772   Germany    Male   42       3   
9999    15628319     Walker          792    France  Female   28       4   

        Balance  Num Of Products  Has Credit Card  Is Active Member  \
9995       0.00                2                1                 0   
9996   57369.61                1                1                 1   
9997       0.00                1                0                 1   
9998   75075.31                2                1                 0   
9999  130142.79                1                1                 0   

      Estimated Salary  Churn  
9995          96270.64      0  
9996         101699.77      0  
9997          42085.58      1  
9998          92888.52      1  
9999          38190.78      0

In [15]:
 # to check the number of rows and columns in a dataframe using shape attribute
churn_data.shape

(10000, 13)

In [16]:
print("Number of Rows",churn_data.shape [0])
print("Number of Columns",churn_data.shape [1])

Number of Rows 10000
Number of Columns 13


In [17]:
churn_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   CustomerId        10000 non-null  int64  
 1   Surname           10000 non-null  object 
 2   CreditScore       10000 non-null  int64  
 3   Geography         10000 non-null  object 
 4   Gender            10000 non-null  object 
 5   Age               10000 non-null  int64  
 6   Tenure            10000 non-null  int64  
 7   Balance           10000 non-null  float64
 8   Num Of Products   10000 non-null  int64  
 9   Has Credit Card   10000 non-null  int64  
 10  Is Active Member  10000 non-null  int64  
 11  Estimated Salary  10000 non-null  float64
 12  Churn             10000 non-null  int64  
dtypes: float64(2), int64(8), object(3)
memory usage: 1015.8+ KB


In [20]:
# checking the null values
#https://www.miamioh.edu/cads/students/coding-tutorials/python/data-cleaning/index.html
churn_data.isnull()

CustomerId  Surname  CreditScore  Geography  Gender    Age  Tenure  \
0          False    False        False      False   False  False   False   
1          False    False        False      False   False  False   False   
2          False    False        False      False   False  False   False   
3          False    False        False      False   False  False   False   
4          False    False        False      False   False  False   False   
...          ...      ...          ...        ...     ...    ...     ...   
9995       False    False        False      False   False  False   False   
9996       False    False        False      False   False  False   False   
9997       False    False        False      False   False  False   False   
9998       False    False        False      False   False  False   False   
9999       False    False        False      False   False  False   False   

      Balance  Num Of Products  Has Credit Card  Is Active Member  \
0       False            False            False             False   
1       False            False            False             False   
2       False            False            False             False   
3       False            False            False             False   
4       False            False            False             False   
...       ...              ...              ...               ...   
9995    False            False            False             False   
9996    False            False            False             False   
9997    False            False            False             False   
9998    False            False            False             False   
9999    False            False            False             False   

      Estimated Salary  Churn  
0                False  False  
1                False  False  
2                False  False  
3                False  False  
4                False  False  
...                ...    ...  
9995             False  False  
9996             False  False  
9997             False  False  
9998             False  False  
9999             False  False  

[10000 rows x 13 columns]

The above .isnull() function gives the boolean value of the dataset

In [21]:
#https://www.miamioh.edu/cads/students/coding-tutorials/python/data-cleaning/index.html
churn_data.isnull().sum()

CustomerId          0
Surname             0
CreditScore         0
Geography           0
Gender              0
Age                 0
Tenure              0
Balance             0
Num Of Products     0
Has Credit Card     0
Is Active Member    0
Estimated Salary    0
Churn               0
dtype: int64

In the above code we are performing sum() of true values to get the null values for both categorical and numerical values. If any missing values are there we should drop them. but here it is not.


In [22]:
churn_data.describe()

CustomerId   CreditScore           Age        Tenure        Balance  \
count  1.000000e+04  10000.000000  10000.000000  10000.000000   10000.000000   
mean   1.569094e+07    650.528800     38.921800      5.012800   76485.889288   
std    7.193619e+04     96.653299     10.487806      2.892174   62397.405202   
min    1.556570e+07    350.000000     18.000000      0.000000       0.000000   
25%    1.562853e+07    584.000000     32.000000      3.000000       0.000000   
50%    1.569074e+07    652.000000     37.000000      5.000000   97198.540000   
75%    1.575323e+07    718.000000     44.000000      7.000000  127644.240000   
max    1.581569e+07    850.000000     92.000000     10.000000  250898.090000   

       Num Of Products  Has Credit Card  Is Active Member  Estimated Salary  \
count     10000.000000      10000.00000      10000.000000      10000.000000   
mean          1.530200          0.70550          0.515100     100090.239881   
std           0.581654          0.45584          0.499797      57510.492818   
min           1.000000          0.00000          0.000000         11.580000   
25%           1.000000          0.00000          0.000000      51002.110000   
50%           1.000000          1.00000          1.000000     100193.915000   
75%           2.000000          1.00000          1.000000     149388.247500   
max           4.000000          1.00000          1.000000     199992.480000   

              Churn  
count  10000.000000  
mean       0.203700  
std        0.402769  
min        0.000000  
25%        0.000000  
50%        0.000000  
75%        0.000000  
max        1.000000

In [23]:
#https://www.w3resource.com/pandas/dataframe/dataframe-describe.php
churn_data.describe(include = 'all')

CustomerId Surname   CreditScore Geography Gender           Age  \
count   1.000000e+04   10000  10000.000000     10000  10000  10000.000000   
unique           NaN    2932           NaN         3      2           NaN   
top              NaN   Smith           NaN    France   Male           NaN   
freq             NaN      32           NaN      5014   5457           NaN   
mean    1.569094e+07     NaN    650.528800       NaN    NaN     38.921800   
std     7.193619e+04     NaN     96.653299       NaN    NaN     10.487806   
min     1.556570e+07     NaN    350.000000       NaN    NaN     18.000000   
25%     1.562853e+07     NaN    584.000000       NaN    NaN     32.000000   
50%     1.569074e+07     NaN    652.000000       NaN    NaN     37.000000   
75%     1.575323e+07     NaN    718.000000       NaN    NaN     44.000000   
max     1.581569e+07     NaN    850.000000       NaN    NaN     92.000000   

              Tenure        Balance  Num Of Products  Has Credit Card  \
count   10000.000000   10000.000000     10000.000000      10000.00000   
unique           NaN            NaN              NaN              NaN   
top              NaN            NaN              NaN              NaN   
freq             NaN            NaN              NaN              NaN   
mean        5.012800   76485.889288         1.530200          0.70550   
std         2.892174   62397.405202         0.581654          0.45584   
min         0.000000       0.000000         1.000000          0.00000   
25%         3.000000       0.000000         1.000000          0.00000   
50%         5.000000   97198.540000         1.000000          1.00000   
75%         7.000000  127644.240000         2.000000          1.00000   
max        10.000000  250898.090000         4.000000          1.00000   

        Is Active Member  Estimated Salary         Churn  
count       10000.000000      10000.000000  10000.000000  
unique               NaN               NaN           NaN  
top                  NaN               NaN           NaN  
freq                 NaN               NaN           NaN  
mean            0.515100     100090.239881      0.203700  
std             0.499797      57510.492818      0.402769  
min             0.000000         11.580000      0.000000  
25%             0.000000      51002.110000      0.000000  
50%             1.000000     100193.915000      0.000000  
75%             1.000000     149388.247500      0.000000  
max             1.000000     199992.480000      1.000000

It displays all the columns of the data frame regardless of its datatype and the overall statistical values of the categorical and numerical columns.

In [7]:
# Dropping inappropriate columns that is not reuired for this analysis
churn_data.columns

Index(['CustomerId', 'Surname', 'CreditScore', 'Geography', 'Gender', 'Age',
       'Tenure', 'Balance', 'Num Of Products', 'Has Credit Card',
       'Is Active Member', 'Estimated Salary', 'Churn'],
      dtype='object')

In [10]:
#https://towardsdatascience.com/3-types-of-data-cleaning-you-must-know-in-python-1ab6986a8b1d
churn_data = churn_data.drop(['CustomerId', 'Surname'], axis=1)

The irrelevant columns are dropped to reduce the memory usage and for the ease of analysis purposes

In [13]:
churn_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 11 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   CreditScore       10000 non-null  int64  
 1   Geography         10000 non-null  object 
 2   Gender            10000 non-null  object 
 3   Age               10000 non-null  int64  
 4   Tenure            10000 non-null  int64  
 5   Balance           10000 non-null  float64
 6   Num Of Products   10000 non-null  int64  
 7   Has Credit Card   10000 non-null  int64  
 8   Is Active Member  10000 non-null  int64  
 9   Estimated Salary  10000 non-null  float64
 10  Churn             10000 non-null  int64  
dtypes: float64(2), int64(7), object(2)
memory usage: 859.5+ KB


After dropping the irrelevant attributes there is a significant decrease in the memory usage from 1015.8+ KB to 859.5+ KB, where the former is the one, before dropping the irrelevant columns.

In [14]:
# Encoding ctaegorical data
churn_data['Geography'].unique()

array(['France', 'Spain', 'Germany'], dtype=object)

In [15]:
# Encoding ctaegorical data
# one hot/ dummy encoding
churn_data = pd.get_dummies(churn_data, drop_first = TRUE)

NameError: name 'TRUE' is not defined